src/recording/macromanager.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <QDebug>
+#include <QObject>
+#include <QtCore>
+
+#include "engine/enginemaster.h"
+#include "mixer/playermanager.h"
+#include "preferences/usersettings.h"
+#include "recording/recordingmanagerbase.h"
+
+#define MACRORECORDING_PREF_KEY "[MacroRecording]"
+
+namespace {
+constexpr size_t kMaxMacroSize = 1000;
+const QLoggingCategory macros("macros");
+}
+
+struct MacroAction {
+    MacroAction(){};
+    MacroAction(double position, double target)
+            : position(position), target(target){};
+    double position;
+    double target;
+};
+
+struct Macro {
+    int m_length = 0;
+    MacroAction actions[kMaxMacroSize];
```

Comment:
Use std::vector or QList rather than unsafe C arrays

Reply:
But it must not be dynamically sized or introduce some other allocation, would that still work?

Reply:
Perhaps QVarLengthArray like ChannelHandleFactory? @daschuer what do you think?

Reply:
std::array?